pydis_site/apps/api/viewsets/bot/nomination.py

```diff
@@ -163,19 +187,50 @@ def create(self, request: HttpRequest, *args, **kwargs) -> Response:
                 raise ValidationError({field: ['This field cannot be set at creation.']})
 
         user_id = request.data.get("user")
-        if Nomination.objects.filter(active=True, user__id=user_id).exists():
-            raise ValidationError({'active': ['There can only be one active nomination.']})
+        nomination_filter = Nomination.objects.filter(active=True, user__id=user_id)
+
+        if not nomination_filter.exists():
+            serializer = NominationSerializer(
+                data=ChainMap(
+                    request.data,
+                    {"active": True}
+                )
+            )
+            serializer.is_valid(raise_exception=True)
+            nomination = Nomination.objects.create(**serializer.validated_data)
 
-        serializer = self.get_serializer(
-            data=ChainMap(
-                request.data,
-                {"active": True}
+            # Serializer truncate unnecessary data away
+            entry_serializer = NominationEntrySerializer(
+                data=ChainMap(request.data, {"nomination": nomination.id})
             )
+            entry_serializer.is_valid(raise_exception=True)
+            NominationEntry.objects.create(**entry_serializer.validated_data)
+
+            data = NominationSerializer(nomination).data
+
+            headers = self.get_success_headers(data)
+            return Response(data, status=status.HTTP_201_CREATED, headers=headers)
+
+        entry_serializer = NominationEntrySerializer(
+            data=ChainMap(request.data, {"nomination": nomination_filter[0].id})
         )
-        serializer.is_valid(raise_exception=True)
-        self.perform_create(serializer)
-        headers = self.get_success_headers(serializer.data)
-        return Response(serializer.data, status=status.HTTP_201_CREATED, headers=headers)
+        entry_serializer.is_valid(raise_exception=True)
+
+        # Don't allow user creating many nomination entries for one nomination
+        if NominationEntry.objects.filter(
+                nomination_id=nomination_filter[0].id,
+                actor__id=entry_serializer.validated_data["actor"].id
+        ).exists():
+            raise ValidationError(
+                {'actor': ['This actor have already created nomination entry for this nomination.']}
```

Comment:
Grammar
```suggestion
                {'actor': ['This actor has already endorsed this nomination.']}
```

Reply:
If this is going to be changed, it's needs to be changed in the test as well